openeo/rest/userfile.py

```diff
@@ -0,0 +1,58 @@
+import typing
+from typing import Any, Dict, List, Union
+from pathlib import Path
+
+if typing.TYPE_CHECKING:
+    # Imports for type checking only (circular import issue at runtime).
+    from openeo.rest.connection import Connection
+
+
+class UserFile:
+    """Represents a file in the user-workspace of openeo."""
+
+    def __init__(self, path: str, connection: 'Connection', metadata: Dict[str, Any] = {}):
+        self.path = path
+        self.metadata = metadata
+        self.connection = connection
+
+    def __repr__(self):
+        return '<{c} file={i!r}>'.format(c=self.__class__.__name__, i=self.path)
+
+    def _get_endpoint(self) -> str:
+        return "/files/{}".format(self.path)
+
+    def get_metadata(self, key) -> Any:
+        """ Get metadata about the file, e.g. file size (key: 'size') or modification date (key: 'modified')."""
+        if key in self.metadata:
```

Comment:
FYI this whole if construct can be a idiomatic oneliner `return self.metadata.get(key)`  (`dict.get()` returns `None` as fallback on missing keys)

Sidenote: attribute `metadata` is public, so added value of `get_metadata()` is unclear in this regard: direct usage is  more intuitive and flexible than using the method:
```
file.metadata["size"]
# vs
file.get_metadata("size")
```

Reply:
Fine for me. Some languages prefer explicit getters/setters and I wasn't sure whether that's the case here.